Bulldozer.CSV/Maps/GroupImport.cs

```diff
@@ -0,0 +1,966 @@
+    {
+        /// <summary>
+        /// Processes the group list.
+        /// </summary>
+        private int ImportGroups( List<GroupCsv> groupCsvList = null, string groupTerm = null )
+        {
+            ReportProgress( 0, "Preparing Group data for import..." );
+
+            if ( this.GroupTypeDict == null )
+            {
+                LoadGroupTypeDict();
+            }
+            if ( this.LocationsDict == null )
+            {
+                LoadLocationDict();
+            }
+
+            LoadGroupDict();
+
+            var groupImportList = new List<GroupImport>();
+            var invalidGroups = new List<string>();
+            var invalidGroupTypes = new List<string>();
+            if ( groupCsvList == null )
+            {
+                groupCsvList = this.GroupCsvList;
+            }
+            if ( groupTerm.IsNullOrWhiteSpace() )
+            {
+                groupTerm = "Group";
+            }
+
+            foreach ( var groupCsv in groupCsvList )
+            {
+                var groupCsvName = groupCsv.Name;
+                if ( groupCsv.Name.IsNullOrWhiteSpace() )
+                {
+                    groupCsvName = $"Unnamed {groupTerm}";
+                }
+                var groupType = this.GroupTypeDict.GetValueOrNull( $"{this.ImportInstanceFKPrefix}^{groupCsv.GroupTypeId}" );
+                if ( groupType == null )
+                {
+                    invalidGroups.Add( groupCsv.Id );
+                    invalidGroupTypes.Add( groupCsv.GroupTypeId );
+                    continue;
+                }
+                var groupImport = new GroupImport()
+                {
+                    GroupForeignId = groupCsv.Id.AsIntegerOrNull(),
+                    GroupForeignKey = $"{this.ImportInstanceFKPrefix}^{groupCsv.Id}",
+                    GroupTypeId = this.GroupTypeDict[$"{this.ImportInstanceFKPrefix}^{groupCsv.GroupTypeId}"].Id,
+                    Name = groupCsvName,
+                    Description = groupCsv.Description,
+                    IsActive = groupCsv.IsActive.GetValueOrDefault(),
+                    IsPublic = groupCsv.IsPublic.GetValueOrDefault(),
+                    Capacity = groupCsv.Capacity,
+                    CreatedDate = groupCsv.CreatedDate,
+                    MeetingDay = groupCsv.MeetingDay,
+                    MeetingTime = groupCsv.MeetingTime,
+                    Order = groupCsv.Order,
+                    ParentGroupForeignId = groupCsv.ParentGroupId.AsIntegerOrNull(),
+                    ParentGroupForeignKey = string.IsNullOrWhiteSpace( groupCsv.ParentGroupId ) ? null : string.Format( "{0}^{1}", ImportInstanceFKPrefix, groupCsv.ParentGroupId )
+                };
+
+                if ( groupCsv.Name.IsNullOrWhiteSpace() )
+                {
+                    groupImport.Name = $"Unnamed {groupTerm}";
+                }
+
+                if ( groupCsv.CampusId.IsNotNullOrWhiteSpace() && groupCsv.CampusId.ToIntSafe( -1 ) != 0 )
```

Comment:
The condition of ToIntSafe( -1 ) != 0 looks weird. Look into and verify why, or change it.